packages/compat/src/build-compat-addon.ts

```diff
@@ -36,16 +42,26 @@ function buildCompatAddon(originalPackage: Package, v1Cache: V1InstanceCache): T
     // because that whole process only depends on looking at all the
     // package.json files on disk -- it can't know which ones are going to end
     // up unused at this point.
-    return new EmptyPackageTree();
+    return { tree: new EmptyPackageTree(), nonResolvableDeps: [] };
+  }
+
+  let nonResolvableDependencies: Set<Package> = new Set();
+  for (let pkg of oldPackages) {
+    for (let dep of pkg.nonResolvableDependencies()) {
+      nonResolvableDependencies.add(dep);
+    }
   }
```

Comment:
@ef4 This is the only spot I am confused about. In your branch you left a comment about how we need to loop over oldPackages and add each ones nonResolvableDependencies. Can you help describe a test case I can add that will have one of the oldPackages contain a different in-repo addon. Right now each oldPackage is basically the same so this is really no different than: `oldPackages[0].nonResolvableDependencies();`

Reply:
I think it's pretty hard to contrive an example where they would differ. I'm OK with not having a test for this, it's enough to review this case and see that it's doing the right thing even if they did differ in the future.